data-serving/data-service/test/model/case.test.ts

```diff
@@ -1,25 +1,35 @@
 import { Case } from '../../src/model/case';
-import fullCase from './data/case.full.json';
-import minimalCase from './data/case.minimal.json';
+import { Error } from 'mongoose';
+import fullModel from './data/case.full.json';
+import minimalModel from './data/case.minimal.json';
 
 describe('validate', () => {
-    it('minimal model should be valid', async () => {
-        const c = new Case(minimalCase);
-
-        return c.validate();
+    it('model without sources is invalid', async () => {
+        return new Case({ ...minimalModel, sources: [] }).validate((e) => {
+            expect(e.name).toBe(Error.ValidationError.name);
+        });
     });
-    it('fully-specified model should be valid', async () => {
-        const c = new Case(fullCase);
 
-        return c.validate();
+    it('model without confirmed event is invalid', async () => {
+        return new Case({ ...minimalModel, events: [] }).validate((e) => {
```

Comment:
nit: this test passes regardless of the `confirmed` name checking

Reply:
good point - separated into 2 tests (empty vs. nonempty, but missing name='sources')